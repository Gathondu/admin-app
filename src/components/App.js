import React, { Component } from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import HomePage from "./home/HomePage";
import AboutPage from "./about/AboutPage";
import Header from "./common/Header";
import Error from "./common/Error";
import CoursesPage from "./course/CoursesPage";
import ManageCoursePage from './course/ManageCoursePage';
import AuthorsPage from "./author/AuthorsPage";
import ManageAuthorPage from "./author/ManageAuthorPage";

export class App extends Component {
    static propTypes = {
        courses: PropTypes.array.isRequired,
        loading: PropTypes.bool.isRequired,
        totalCourses: PropTypes.number.isRequired,
        totalAuthors: PropTypes.number.isRequired
    };

    render() {
        const { loading, totalCourses, totalAuthors} = this.props;
        return (
            <BrowserRouter>
                <div className="container">
                    <Header loading={loading} totalCourses={totalCourses} totalAuthors={totalAuthors}/>
                    <Switch>
                        <Route exact path="/" component={HomePage}/>
                        <Route path="/authors" render={(props) => <AuthorsPage {...props}/>}/>
                        <Route exact path="/author" render={(props) => <ManageAuthorPage {...props}/>}/>
                        <Route exact path="/author/:id" render={(props) => <ManageAuthorPage {...props}/>}/>
                        <Route path="/courses" render={(props) => <CoursesPage {...props} />}/>
                        <Route exact path="/course" render={(props) => <ManageCoursePage {...props}/>}/>
                        <Route exact path="/course/:id" render={(props) => <ManageCoursePage {...props}/>}/>
                        <Route path="/about" component={AboutPage}/>
                        <Route component={Error}/>
                    </Switch>
                </div>
            </BrowserRouter>
        );
    }
}
export const mapStateToProps = state => (
    {
        courses: state.courses,
        authors: state.authors,
        loading: state.ajaxCallsInProgress > 0,
        totalCourses: state.totalCourses,
        totalAuthors: state.totalAuthors
    }
);

export default connect(mapStateToProps)(App);
