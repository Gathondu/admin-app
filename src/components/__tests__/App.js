import React from 'react';
import {mount} from "enzyme";
import HomePage from "../home/HomePage";
import {App} from "../App";
import {MemoryRouter} from "react-router-dom";
import configureStore from "../../store";
import {Provider} from "react-redux";

const store = configureStore();
const renderRoutes = path =>
    mount(
        <MemoryRouter initialEntries={[path]}>
            <Provider store={store}>
                <App totalCourses={4} totalAuthors={3} loading={false} courses={[]}/>
            </Provider>
        </MemoryRouter>
    );

describe('App component', () => {
    it('should mount in a full DOM', () => {
        const wrapper = mount(
            <App
                courses={[]}
                loading={false}
                totalCourses={4}
                totalAuthors={4}/>
        );
        expect(wrapper.find('.container').length).toBe(1);
    });
    it('should render / route correctly', () => {
        const component = renderRoutes('/');
        expect(component.find(HomePage)).toHaveLength(1)
    });
});
