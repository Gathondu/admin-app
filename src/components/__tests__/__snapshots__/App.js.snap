// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without crashing 1`] = `
<BrowserRouter>
  <div
    className="container"
  >
    <Header
      loading={false}
      totalAuthors={3}
      totalCourses={4}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        path="/authors"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/author"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/author/:id"
        render={[Function]}
      />
      <Route
        path="/courses"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/course"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/course/:id"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
